plugins/org.eclipse.fordiac.ide.fb.interpreter/src/org/eclipse/fordiac/ide/fb/interpreter/DefaultRunFBType.java

```diff
@@ -316,12 +319,31 @@ public EList<EventOccurrence> runSimpleFBType(final SimpleFBTypeRuntime simpleFB
 		// Initialization of variables
 		final SimpleFBType simpleFBType = simpleFBTypeRuntime.getSimpleFBType();
 		VariableUtils.fBVariableInitialization(simpleFBType);
-		final var outputEvents = new BasicEList<EventOccurrence>();
-		processAlgorithmWithEvaluator(simpleFBType, simpleFBType.getAlgorithm().get(0), eventOccurrence);
+
+		final List<SimpleECAction> actions = getActions(simpleFBType, eventOccurrence.getEvent().getName());
+
+		for (final SimpleECAction action : actions) {
+			simpleFBType.getAlgorithm().stream().filter(a -> a.getName().equals(action.getAlgorithm())).findAny()
```

Comment:
instead of the stream it should be possible that you just take the algorithm that is refered to by the action, if not null.

the event of the action could be put into a return list. This would reduce to have the stream at the bottom

Reply:
ah sorry I forgot that. Thx for the clarfication